apps/emqx/src/emqx_tls_lib.erl

```diff
@@ -426,6 +430,51 @@ file_content_as_options([Key | Keys], SSL) ->
             end
     end.
 
+%% @doc Convert hocon-checked ssl client options (map()) to
+%% proplist accepted by ssl library.
+to_client_opts(Opts) ->
+    GetD = fun(Key, Default) -> fuzzy_map_get(Key, Opts, Default) end,
+    Get = fun(Key) -> GetD(Key, undefined) end,
+    KeyFile = ensure_str(Get(keyfile)),
+    CertFile = ensure_str(Get(certfile)),
+    CAFile = ensure_str(Get(cacertfile)),
```

Comment:
Just noting: these will break if they are not defined, since `ensure_str` does not accept atoms.